src/HDF5.jl

```diff
@@ -1181,16 +1181,25 @@ function Base.read(obj::DatasetOrAttribute, ::Type{T}, I...) where T
         end
     end
 
-    buf = Array{T}(undef, sz...)
-    memspace = dataspace(buf)
+    sz = Int.(sz)
```

Comment:
rebase and get rid of this?

Reply:
Yeah, I can do that. I'll get this and #745 rebased.